avocado/utils/asset.py

```diff
@@ -178,3 +192,13 @@ def _check_file(path, filehash=None, algorithm=None):
             log.error('Asset %s corrupted (hash expected:%s, hash found:%s).' %
                       (path, filehash, discovered_hash))
             return False
+
+    @staticmethod
+    def _is_expired(path, expire):
+        if expire is None or expire < 0:
+            return False
+        creation_time = os.lstat(path)[stat.ST_CTIME]
```

Comment:
well as the file could be just a symlink, then I'd suggest using `os.realpath(path)` to get the original file's creation time.